src/Maps/MapType.h

```diff
@@ -0,0 +1,6 @@
+#pragma once
+
+enum class MapType {
+	Map,
+	Save
+};
```

Comment:
I believe when an `enum` is this closely tied to a class, it is appropriate to declare it in the class header file, and probably within the namespace of the class itself, as a public member. Everything you need related to the class is then in one place.

For a really long enum, it may make sense to put it in a separate file. For an enum used by several classes with no clear owner, it definitely makes sense to put it in a separate file. This case I think is optional.

Reply:
I'm happy either way embedding in MapData or not. I was more worried about trying to find a better name than MapType, which I thought could be improved.

Reply:
The enum name seems alright. Even if it was embedded, I think it would still be fine. I maybe feel "Save" is a bit short though, and would prefer "SavedGame". The reason being that "Save" is a verb, whereas the other two are nouns.

Reply:
Yeah, save is pretty generic, good call.